src/index.test.ts

```diff
@@ -183,17 +195,46 @@ test('Should include context fields on request', async () => {
     const url = new URL(fetchMock.mock.calls[0][0]);
 
     expect(url.searchParams.get('userId')).toEqual('123');
-    expect(url.searchParams.get('randomField')).toEqual('random');
+    expect(url.searchParams.get('sessionId')).toEqual('456');
+    expect(url.searchParams.get('remoteAddress')).toEqual('address');
+    expect(url.searchParams.get('properties.property1')).toEqual('property1');
+    expect(url.searchParams.get('properties.property2')).toEqual('property2');
+    expect(url.searchParams.get('appName')).toEqual('web');
+    expect(url.searchParams.get('environment')).toEqual('prod');
+});
+
+test('Should not add property fields when properties is an empty object', async () => {
+    fetchMock.mockResponses(
+        [JSON.stringify(data), { status: 200 }],
+        [JSON.stringify(data), { status: 304 }],
+    );
+    const config: IConfig = {
+        url: 'http://localhost/test',
+        clientKey: '12',
+        context: { appName: 'web', environment: 'prod' },
+    };
+    const client = new UnleashClient(config);
+    client.updateContext({
+        properties: {}
+    });
+
+    await client.start();
+
+    jest.advanceTimersByTime(1001);
+
+    const url = new URL(fetchMock.mock.calls[0][0]);
+
     expect(url.searchParams.get('appName')).toEqual('web');
     expect(url.searchParams.get('environment')).toEqual('prod');
+    expect(url.searchParams.get('properties')).toBeNull();
```

Comment:
Should it be null or undefined?

Reply:
Good question. I also thought the same thing, but it comes back null when it's not in the search params